cosima_cookbook/explore.py

```diff
@@ -898,7 +900,7 @@ def __init__(self, session=None, experiment=None):
             expts = querying.get_experiments(self.session, all=True)
```

Comment:
This logic probably needs to be adjusted after pushing `session` handling into the `DatabaseExtension` too?

Reply:
Bugger. Maybe this is what prompted moving away from doing all this in `DatabaseExtension`. Rings a bell now. I modified `DatabaseExtension` to take a list of experiments when I needed to be able to use it for filtering as well as just choosing a variable in ExperimentExplorer. 

I think I'll revert to doing the session open in the two routines and take it out of `DatabaseExplorer`.

Reply:
Thanks for spotting that BTW.